packages/frontend/src/ee/features/customRoles/useCustomRole.ts

```diff
@@ -0,0 +1,133 @@
+import {
+    type AddScopesToRole,
+    type ApiError,
+    type Role,
+    type RoleWithScopes,
+    type UpdateRole,
+} from '@lightdash/common';
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query';
+import { useMemo } from 'react';
+import { lightdashApi } from '../../../api';
+import { useOrganization } from '../../../hooks/organization/useOrganization';
+import useToaster from '../../../hooks/toaster/useToaster';
+
+const CACHE_KEY = 'custom-role';
+
+/**
+ * Hook for fetching and updating a single custom role
+ * @param roleUuid - Optional role UUID for automatic fetching
+ * @param options - Query options
+ * @returns Query result and update mutation
+ */
+export const useCustomRole = (roleUuid?: string) => {
+    const queryClient = useQueryClient();
+    const { showToastApiError, showToastSuccess } = useToaster();
+    const { data: organization } = useOrganization();
+
+    const query = useQuery<RoleWithScopes, ApiError>({
+        queryKey: [CACHE_KEY, roleUuid],
+        queryFn: async () => {
+            if (!roleUuid) {
+                throw new Error('Role UUID is required');
+            }
+            return lightdashApi<RoleWithScopes>({
+                method: 'GET',
+                url: `/roles/${roleUuid}`,
+                version: 'v2',
+            });
+        },
+        enabled: !!roleUuid,
+    });
+
+    const updateRole = useMutation<
+        Role,
+        ApiError,
+        {
+            roleUuid: string;
+            data: UpdateRole & {
```

Comment:
we could simplify this by sending the full list of scopes, and do a POST instead of a PATCH.